File: test/5.py
def solution(A):
    start = 0
    end = start + 1
    longest = 0

    A.sort()
    while end <= len(A):
        tmp = A[start:end]
        if amplitude(tmp) < 2:
            end += 1
            if len(tmp) > longest:
                longest = len(tmp)
        else:
            start = end
            end = start + 1
    return longest

def amplitude(tmp):
    if len(tmp) < 2:
        return 0
    else:
        return max(tmp) - min(tmp)

File: test/5.go
package test

import (
	"fmt"
	"sort"
)

func Solution5() {
	arr := []int{6, 10, 6, 9, 7, 8}
	amp := solution5(arr)
	fmt.Println("amplitude", amp)
}

func solution5(A []int) int {
	s := 0
	e := s + 1
	longest := 0

	sort.Ints(A)
	for e <= len(A) {
		sub := A[s:e]
		if maxAmp(sub) < 2 {
			e++
			if longest > len(sub) {
				longest = len(sub)
			}
		} else {
			s = e
			e = s + 1
		}
	}
	return longest
}

func maxAmp(sub []int) int {
	if len(sub) < 2 {
		return 0
	}
	return sub[len(sub)-1] - sub[0]
}
